pkg/mls/api/v1/service.go

```diff
@@ -88,28 +87,18 @@ func (s *Service) Close() {
 
 func (s *Service) HandleIncomingWakuRelayMessage(wakuMsg *wakupb.WakuMessage) error {
 	if topic.IsMLSV1Group(wakuMsg.ContentTopic) {
-		var msg mlsv1.GroupMessage
-		err := pb.Unmarshal(wakuMsg.Payload, &msg)
-		if err != nil {
-			return err
-		}
-		if msg.GetV1() == nil {
-			return nil
-		}
-		err = s.nc.Publish(buildNatsSubjectForGroupMessages(msg.GetV1().GroupId), wakuMsg.Payload)
+		// The waku message payload is just the group ID as bytes since we only
+		// need to use it as a signal that a new message was published, without
+		// any other content.
+		err := s.nc.Publish(buildNatsSubjectForGroupMessages(wakuMsg.Payload), wakuMsg.Payload)
 		if err != nil {
 			return err
 		}
 	} else if topic.IsMLSV1Welcome(wakuMsg.ContentTopic) {
-		var msg mlsv1.WelcomeMessage
-		err := pb.Unmarshal(wakuMsg.Payload, &msg)
-		if err != nil {
-			return err
-		}
-		if msg.GetV1() == nil {
-			return nil
-		}
-		err = s.nc.Publish(buildNatsSubjectForWelcomeMessages(msg.GetV1().InstallationKey), wakuMsg.Payload)
+		// The waku message payload is just the installation key as bytes since
+		// we only need to use it as a signal that a new message was published,
+		// without any other content.
+		err := s.nc.Publish(buildNatsSubjectForWelcomeMessages(wakuMsg.Payload), wakuMsg.Payload)
```

Comment:
One thing that is a hard requirement here is some way of supporting `streamAllMessages`, since we know integrators are going to want to use their push servers with MLS. Is the idea that each `streamAllMessages` connection would poll the database for any new messages that have arrived since the last time it checked?

Reply:
Yep SubscribeAll logic would look similar but without the group/installation scoping. It could use the nats wildcard topic as signal for activity, but not sure if we'd really need the signal in the all case vs just assuming there's always activity. I think it'd probably be reasonable to have a slightly larger ticker period for SubscribeAll too if we want to tune it a bit for that case